### ISSUE TITLE
Extend LOAD_KEYS_CASE_INSENSITIVELY to support something like snake_case

### ISSUE THREAD

**Issue by Monchi** [uchuhimo/konf#40]
Please think about following example code.
```kotlin
object ExampleSpec : ConfigSpec("") {
    private val exampleKey by required<String>()
    
    fun loadExampleKey(): String {
        val config = Config { addSpec(ExampleSpec) }
        	.enable(Feature.LOAD_KEYS_CASE_INSENSITIVELY)
            .withSource(
            	Source.from.json.file(File("config.json"))
            )
		
        return config[exampleKey]
    }
}
```

With the code above, We can load:
```json
{
  "exampleKey": "value"
}
```
But we cannot load with error:
`Exception in thread "main" java.util.concurrent.ExecutionException: com.uchuhimo.konf.UnsetValueException: item indentStyle is unset`
```json
  "example_key": "value"
```

It's very nice if both CamelCase and snake_case are supported, especially when we load config from multiple file type such as json and yaml.

**Comment by uchuhimo** [comment]
@Monchi What's your preferred solution? What about adding a pair of functions `source.toSnakeCase()` and `source.toCamelCase()` to convert between these two conventions?

P.S. You can explicitly specify the name to workaround it now:

```kotlin
private val exampleKey by required<String>("example_key")
```

**Comment by Monchi** [comment]
@uchuhimo 
I want to present users options to use snake case or camel case. So I need `example_key` and `exampleKey` to be mapped to `config[exampleKey]` implicitly.

**Comment by uchuhimo** [comment]
@Monchi I got it. I will consider implementing it in the next release. A pull request is welcome if you are willing to contribute. Since I maintain Konf part-time, there may be some delay before the next release.

**Comment by uchuhimo** [comment]
@Monchi I have released a new version v0.23.0 that implements the desired behaviour. You can try it now. Any further feedback is welcome.
